src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyNamingPolicy.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json.Serialization
+{
+    /// <summary>
+    /// Determines the policy used to convert a JSON name to another format, such as a camel-casing format.
+    /// </summary>
+    public abstract class JsonPropertyNamingPolicy
```

Comment:
Why did we change this? The naming policy could be applied to dictionary keys. In the future it might also apply to enum string values. `JsonPropertyNamingPolicy` isn't specific to properties.

Reply:
Agree. cc @rynowak

Reply:
> In the future it might also apply to enum string values.

Really? Do you think that is something we would consider extending this type to? That seems a bit odd to me.

My concern with `JsonNamingPolicy` was that it was too generic/ambiguous, though if you feel strongly about it, I can withdraw my concern. cc @terrajobst

Reply:
`JsonNamingPolicy` seems right. We have plans to use it for dictionary keys as well.

Reply:
> Really? Do you think that is something we would consider extending this type to? That seems a bit odd to me.

Think of an enum string as the name for a value. It fits 😄